test/lib/test.js

```diff
@@ -4,15 +4,39 @@ const λ = require('fantasy-check/src/adapters/nodeunit');
 const applicative = require('fantasy-check/src/laws/applicative');
 const functor = require('fantasy-check/src/laws/functor');
 const monad = require('fantasy-check/src/laws/monad');
-    
+const helpers = require('fantasy-check/src/laws/helpers');
+
 const daggy = require('daggy');
 
 const {isInstanceOf} = require('fantasy-helpers');
-const {constant, identity} = require('fantasy-combinators');
+const {constant, identity, compose} = require('fantasy-combinators');
 
 const Identity = require('fantasy-identities');
 const State = require('../../fantasy-states');
 
+const equality = helpers.equality;
+
+const mfunctor = {
+    composition: function(λ) {
+        return function(T, unpack) {
+            return λ.check(function(a) {
+                var x = T.hoist(compose(identity)(identity))(T.of(a)),
```

Comment:
It would be ideal to use monad morphisms, instead of identity: `IdToList`, `ListToMaybe` or something, but i think it's ok to leave as is for now.